inductiva/_benchmarks/benchmark.py

```diff
@@ -133,38 +141,78 @@ def wait(self) -> Self:
+        """
+        Gathers information, including metrics and metadata, from all tasks.
+
+        Args:
+            verbose (bool): A boolean indicating whether to include additional
+                details for each task in the output (default is False).
 
         Returns:
-            dict: A dictionary containing the results of each task, organized 
-            by virtual machine type.
+            dict: A dictionary containing the details of each task,
+                organized by virtual machine type.
         """
-        results = {}
+
+        def _gather_verbose(data: dict, info: TaskInfo, vm_type: str):
+            INPUT_FILENAME = "input.json"
+            input_dir_path = task.download_inputs(filenames=[INPUT_FILENAME])
+            input_file_path = input_dir_path.joinpath(INPUT_FILENAME)
+            with open(input_file_path, mode="r", encoding="utf-8") as file:
+                input_json = json.load(file)
+            data[vm_type].append({
+                "task_info": info.to_dict(),
+                "task_input_metadata": input_json,
+            })
+
+        def _gather_summary(data: dict, info: TaskInfo, vm_type: str):
+            data[vm_type].append({
+                "task_id": info.task_id,
+                "estimated_computation_cost": info.estimated_computation_cost,
+                "computation_time": info.time_metrics.computation_seconds.value,
+            })
+
+        data = {}
         tasks = self.get_tasks()
         for task in tasks:
-            task_info = task.info
-            tmp1, tmp2, *_ = task_info.executer.vm_name.split("-")
-            machine_group_name = tmp1 + "-" + tmp2
-            resource = machine_groups.get_by_name(machine_group_name)
-            vm_type = task_info.executer.vm_type
-            if vm_type not in results:
-                cost_per_hour = resource.estimate_cloud_cost(verbose=False)
-                results[vm_type] = {
-                    "cost per hour": cost_per_hour,
-                    "machine type": resource.__class__.__name__,
-                    "provider": task_info.executer.host_type,
-                    "machine count": resource.num_machines,
-                    "tasks": []
-                }
-            metadata_filename = "input.json"
-            inputs_path = task.download_inputs(filenames=[metadata_filename])
-            metadata_path = inputs_path.joinpath(metadata_filename)
-            with open(metadata_path, mode="r", encoding="utf-8") as file:
-                metadata_content = json.load(file)
-            results[vm_type]["tasks"].append({
-                "info": task_info.to_dict(),
-                "metadata": metadata_content,
-            })
-        return results
+            info = task.get_info()
+            vm_type = info.executer.vm_type
+            data.setdefault(vm_type, [])
+            gather_function = _gather_verbose if verbose else _gather_summary
+            gather_function(data, info, vm_type)
```

Comment:
I'm not a big fan of using parameters as input/output.
Since the _gather functions are internal and dont have docstrings is not that clear what is going to happen.

Maybe make the gather_funtion return one record and append that in the main function.

This is just personal preference.

Reply:
I'm not sure I fully understand the part about 'using parameters as input/output'. Could you clarify a bit more?

Reply:
Your data is an input of the method gather_function but it also works as an output since you modify data inside the method.
You can do that
```python
#Data gets modified inside gather_function
gather_function(data, info, vm_type)
```
Or
```python
#you have your data
new_data = gather_function(info)

data[vm_type].append(new_data)
```

I just prefer the second one since inputs and outputs are clearly defined. But its like I said before. It's just personal preference.

Reply:
I get what you’re saying. This function is being refactored.